File: src/main/java/cabalgatas_salento/reservas/mapper/RutaMapper.java
package cabalgatas_salento.reservas.mapper;

import cabalgatas_salento.reservas.dto.request.RutaRequest;
import cabalgatas_salento.reservas.dto.response.RutaResponse;
import cabalgatas_salento.reservas.entity.Ruta;
import org.springframework.stereotype.Component;

@Component
public class RutaMapper {

    public RutaResponse toResponse(Ruta ruta) {
        return RutaResponse.builder()
                .id(ruta.getId())
                .nombre(ruta.getNombre())
                .descripcion(ruta.getDescripcion())
                .precio(ruta.getPrecio())
                .dificultad(ruta.getDificultad())
                .duracionMinutos(ruta.getDuracionMinutos())
                .imageUrl(ruta.getImageUrl())
                .isActive(ruta.getIsActive())
                .eliminado(ruta.getEliminado())
                .build();
    }

    public Ruta toEntity(RutaRequest request) {
        return Ruta.builder()
                .nombre(request.getNombre())
                .descripcion(request.getDescripcion())
                .precio(request.getPrecio())
                .dificultad(request.getDificultad())
                .duracionMinutos(request.getDuracionMinutos())
                .imageUrl(request.getImageUrl())
                .build();
    }
}

File: src/rutas/mappers/ruta.mapper.ts
import { Ruta } from '../entities/ruta.entity';
import { RutaResponseDto } from '../dto/ruta-response.dto';

export class RutaMapper {
  static toResponseDto(entity: Ruta): RutaResponseDto {
    const dto = new RutaResponseDto();
    dto.id = entity.id;
    dto.nombre = entity.nombre;
    dto.descripcion = entity.descripcion;
    dto.precio = entity.precio;
    dto.dificultad = entity.dificultad;
    dto.duracionMinutos = entity.duracionMinutos;
    dto.imageUrl = entity.imageUrl;
    dto.isActive = entity.isActive;
    dto.eliminado = entity.eliminado;
    return dto;
  }
}
